wpcom-legacy-redirector.php

```diff
@@ -170,7 +221,15 @@ static function get_redirect_post_id( $url ) {
 
 		$url_hash = self::get_url_hash( $url );
 
-		$redirect_post_id = $wpdb->get_var( $wpdb->prepare( "SELECT ID FROM $wpdb->posts WHERE post_type = %s AND post_name = %s LIMIT 1", self::POST_TYPE, $url_hash ) );
+		// Allow plugins to disable lowercase. Check in case we transform to lowercase.
+		if ( apply_filters( 'wpcom_legacy_redirector_check_lowercase', true ) ) {
```

Comment:
Should this be a separate feature / PR?

Reply:
I'm not sure, to be honest. This was part of the original branch.  Do you see any benefits to always returning a lowercase URL?

I don't really see any added benefits, maybe older versions of WordPress had a problem with this?  I think we should probably remove it unless you see a reason to keep it.